list: fix --trashed to return all items in the current working directory

njbbaer opened the pull request:
Fixes #695.

odeke-em commented:
Apologies for the late reply, I'll comment more on this PR in about 9 hours when I get back home.

odeke-em commented on src/remote.go:
```diff
@@ -142,11 +142,7 @@ func (r *Remote) changes(startChangeId int64) (chan *drive.Change, error) {
 func buildExpression(parentId string, typeMask int, inTrash bool) string {
 	var exprBuilder []string
 
-	if inTrash || (typeMask&InTrash) != 0 {
```
Aha I see, so this simplifies the trashed condition checks. Does it work alright with `list` as well as `--trashed` in nested directories, not just root?

njbbaer replied:
Yes, `list` and `list --trashed` now work in both root and nested directories. Please test this yourself.

What broke `list --trashed` was two separate small bugs. First, `list --trashed` wasn't working in nested directories because it was looking for the directory in trash rather than just its containing files in trash. So I switched `list` to use its default path resolver `FindByPath` instead of `FindByPathTrashed`.

Second, `list --trashed` returned all trashed items in all directories. I found in `buildExpression` that a parent directory was not specified if searching `inTrash`. Adding it back to match the regular case fixed the issue. I also removed the `(typeMask&InTrash) != 0` check because I felt it shouldn't contradict the `inTrash` parameter and simplified the expression even more.

njbbaer commented:
What broke `list --trashed` was two separate small bugs. First, it wasn't working in nested directories because it was looking for the directory in trash rather than just its containing files in trash. So I switched `list` to use its default path resolver `FindByPath` instead of `FindByPathTrashed`.

Second, `list --trashed` returned all trashed items in all directories. I found in `buildExpression` that a parent directory was not specified if searching `inTrash`. Adding it back to match the regular case fixed the issue. I also removed the `(typeMask&InTrash) != 0` check because I felt it shouldn't contradict the `inTrash` parameter and simplified the expression even more.

odeke-em commented:
Dope! Thanks @njbbaer. Just one comment here, would you mind updating your commit message to reflect the change with something like this

```shell
list: fix --trashed to return all items in the current working directory

Fixes #695

<Any description of the solution if you'd like to make one>
```

njbbaer commented:
Thanks for the tip. I've updated my commit message.

odeke-em commented:
Oh, my bad, I meant in the actual git commit like this
```shell
$ git commit --amend
```
and then when it opens up the editor
```shell
list: fix --trashed to return all items in the current working directory

Fixes #695

<Any description of the solution if you'd like to make one>
```
and after force push it to overwrite this current commit.
```shell
$ git push origin <branch_name> -f
```

njbbaer commented:
That's what I did. I changed both the commit message and the pull request title.

odeke-em commented:
Awesome. Thank you. LGTM! Merging in.
